clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -21703,32 +21703,37 @@ const ValueDecl *SemaOpenMP::getOpenMPDeclareMapperVarName() const {
   return cast<DeclRefExpr>(DSAStack->getDeclareMapperVarRef())->getDecl();
 }
 
-OMPClause *SemaOpenMP::ActOnOpenMPNumTeamsClause(Expr *NumTeams,
+OMPClause *SemaOpenMP::ActOnOpenMPNumTeamsClause(ArrayRef<Expr *> VarList,
                                                  SourceLocation StartLoc,
                                                  SourceLocation LParenLoc,
                                                  SourceLocation EndLoc) {
-  Expr *ValExpr = NumTeams;
-  Stmt *HelperValStmt = nullptr;
-
-  // OpenMP [teams Constrcut, Restrictions]
-  // The num_teams expression must evaluate to a positive integer value.
-  if (!isNonNegativeIntegerValue(ValExpr, SemaRef, OMPC_num_teams,
-                                 /*StrictlyPositive=*/true))
+  if (VarList.empty())
```

Comment:
@alexey-bataev Is there any approach here to check the existence of other clause of the current directive?

My goal here is, if `ompx_bare` clause exists, `num_teams` clause can accept multiple expressions; otherwise it should only accept one.

Reply:
Here no, you need to add a check in the whole directive processing function (check checkSimdlenSafelenSpecified and some other similar)

Reply:
So basically we can only check whether those values are valid but in terms of how many values it can allow, now it will have to defer to the `ActOnXXXDirective` function right?

Reply:
Yes